stdlib/LinearAlgebra/src/generic.jl

```diff
@@ -647,6 +647,136 @@ julia> norm(-2, Inf)
 end
 norm(::Missing, p::Real=2) = missing
 
+# With dims keyword
+norm0_dims!(B, A) = count!(!iszero, B, A)
+norm1_dims!(B, A) = Base.mapreducedim!(norm, +, B, A)
+normInf_dims!(B, A) = Base.mapreducedim!(norm, max, B, A)
+normMinusInf_dims!(B, A) = Base.mapreducedim!(norm, min, B, A)
```

Comment:
BTW the reason these are all mutating `B` is to work around #43461. With things like `mapreduce(norm, max, A; dims)` instead, some were not type-stable. There's a PR to fix that, though.